modules/swagger-codegen/src/main/java/io/swagger/codegen/SupportingFile.java

```diff
@@ -44,4 +48,17 @@ public int hashCode() {
         result = 31 * result + (destinationFilename != null ? destinationFilename.hashCode() : 0);
         return result;
     }
-}
\ No newline at end of file
+
+    static public class GlobalSupportingFile extends SupportingFile {
```

Comment:
I treat this as a integral part of the SupportingFile but I can move it to separate place.